### ISSUE TITLE
Nemo content search - fills all ram

### ISSUE THREAD

**Issue by Navy-Ro** [linuxmint/nemo#3472]
### Distribution

current LMDE6 & Ubu - LM22

### Package version

6.2.9 on both

### Graphics hardware in use

Lmde: Intel Corporation CoffeeLake-S GT2 [UHD Graphics 630] & /Ubu: an old amd saphire

### Frequency

Always

### Bug description

Hi, every time i do a (personal number) **Content Search with Nemo in Home directory or Docs** (no other apps runing) the RAM starts to fill up to maximum till PC stops responding. (it also happened in previsious versions LM & LMDE - than i used recoll wich i dont prefer)
!
(in the screen capture below i closed Cinnamon before it locked my pc)
[Captură de ecran din 2024-10-19 03-29-40](https://github.com/user-attachments/assets/9c585868-cee3-4665-9b99-a8a24fec9968)

### Steps to reproduce

simply do a **Content Search with Nemo in Home directory** - too much files maybe ? :)

### Expected behavior

--

### Additional information

maybe an **Optional file indexer AddOn / Nemo** (like tracker/baloo) - would be better for someone managing lots of docs/files.
Thank u all LM Team !
![Captură de ecran din 2024-10-19 04-42-20](https://github.com/user-attachments/assets/753b6353-95dc-4c8c-b98b-2c054ec26fa2)

**Comment by Jeremy7701** [comment]
Tried to reproduce this with a content search for 'abc' in my home directory.
Took 4m 30secs - over about 60,000 files, with other apps running without any problems.
```
               total        used        free      shared  buff/cache   available
Mem:            31Gi       3.2Gi        23Gi        28Mi       5.2Gi        27Gi
```

**Comment by Navy-Ro** [comment]
![Captură de ecran din 2024-10-22 17-15-40](https://github.com/user-attachments/assets/cea23536-ec96-4b94-a920-aebcdb1243d2)
![Captură de ecran din 2024-10-22 17-16-38](https://github.com/user-attachments/assets/bc72b548-7261-406d-ba04-6da26c50bccb)
![Captură de ecran din 2024-10-22 17-18-53](https://github.com/user-attachments/assets/c09f7a94-8896-4345-a442-ffed938f63c1)
![Captură de ecran din 2024-10-22 17-20-34](https://github.com/user-attachments/assets/e16e5c39-1d57-4990-a88d-54563d274778)
![Captură de ecran din 2024-10-22 17-47-15](https://github.com/user-attachments/assets/0319b100-fd4c-4030-9e82-03b1561759a5)

 Strange behavior in Debug mode, no error, no fully locked pc, 
but at some times (aprox. times of screenshots - when i could move the mouse) stopped/paused on some docs, filled memory up to maximum, locks pc for 5-10 seconds, than without any user action empty/returns RAm to normal usage and continues search till end of files. (33 minutes / 22.2 gigs of archived files - last screenshot).
 Tomorrow i will try on other LM (old machine).
THX!

**Comment by Navy-Ro** [comment]
Hi, searched again and looked at every file it stopped/paused to find those files were broken/corrupted.
it was complaining about realpath like in foto,
!
[Captură de ecran din 2024-10-24 01-47-48](https://github.com/user-attachments/assets/d43a3337-9932-4409-b259-cb0cbb4a9250)
deleted those files and now all seems to run fine (except the time it takes).
Thank u!

**Comment by mtwebster** [comment]
Strange - are those old MS Word .doc files?  Do you have others of that format that are searched successfully?

We use a 3rd party program (catdoc) to content search that file type, I wonder if that was what's failing.

Are those files gone completely? Maybe you could just run `catdoc <filename` to see if that also fails/fills ram. I could probably limit its resources and exit if it uses too much, instead of what you experienced.

Either way, glad you got it resolved

**Comment by Navy-Ro** [comment]
yes, old MS Word .doc files, i tottaly deleted that files on this pc,
but i surely find'em again, maybe monday, on that Backup stick at the office.
i will come back with an answer ...

**Comment by Navy-Ro** [comment]
sorry for late resp.
![Captură de ecran din 2024-11-13 19-25-06](https://github.com/user-attachments/assets/cf395421-aad7-49e6-bfb4-d0f9bffbe8c4)
catdoc does the same, fills memory and when to max it kills the process.
thx
